crates/router/src/core/payment_methods/cards.rs

```diff
@@ -1784,11 +1801,38 @@ pub async fn list_customer_payment_method(
     for pm in resp.into_iter() {
         let parent_payment_method_token = generate_id(consts::ID_LENGTH, "token");
         let hyperswitch_token = generate_id(consts::ID_LENGTH, "token");
-        let card = if pm.payment_method == enums::PaymentMethod::Card {
-            Some(get_lookup_key_from_locker(state, &hyperswitch_token, &pm).await?)
-        } else {
-            None
-        };
+
+        let card: Option<api::CardDetailFromLocker> =
+            if pm.payment_method == enums::PaymentMethod::Card {
```

Comment:
Write card related things as a separate function and call that here.

Reply:
Have done